cms/models.py

```diff
@@ -12,6 +13,63 @@
 from wagtail.images.blocks import ImageChooserBlock
 
 
+class CourseProgramChildPage(Page):
+    """
+    Abstract page representing a child of Course/Program Page
+    """
+
+    class Meta:
+        abstract = True
+
+    parent_page_types = ["CoursePage", "ProgramPage"]
+
+    # disable promote panels, no need for slug entry, it will be autogenerated
+    promote_panels = []
+
+    @classmethod
+    def can_create_at(cls, parent):
+        # You can only create one of these page under course / program.
+        return (
+            super(CourseProgramChildPage, cls).can_create_at(parent)
+            and parent.get_children().type(cls).count() == 0
+        )
+
+
+class ForTeamsPage(CourseProgramChildPage):
+    """
+    CMS Page representing a "For Teams" section in a course/program page
+    """
+
+    content = RichTextField(help_text="The content shown in the section")
+    action_title = models.CharField(
+        max_length=255, help_text="The text to show on the call to action button"
+    )
+    switch_layout = models.BooleanField(
+        blank=True,
+        help_text="When checked, switches the position of the image and content, i.e. image on left and content on right.",
+    )
+    image = models.ForeignKey(
+        Image,
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+        help_text="Image size must be at least 750x505 pixels.",
+    )
+    content_panels = [
+        FieldPanel("content"),
+        FieldPanel("action_title"),
+        FieldPanel("switch_layout"),
+        FieldPanel("image"),
+    ]
+
+    def save(self, *args, **kwargs):
+        # autogenerate a unique slug so we don't hit a ValidationError
+        self.title = "For Teams"
+        self.slug = slugify("{}-{}".format(self.get_parent().id, self.title))
```

Comment:
what will `self.get_parent().id` return ? how parent will be defined ?

Reply:
Will return the `pk` of parent page. This subpage type is restricted to only be created under a `CoursePage` or `ProgramPage`. The parent is already available by the time `save` is called.

Reply:
do we need this method for every subtype ?

Reply:
Yes, need to have a system-wide unique slug for each wagtail page.